### PR TITLE
Create page elements explicitly with typescript static methods.

### PR DESCRIPTION
**Pull request by maffoo**
This has the advantage that we control when elements are created (so that elements do not share state between page transitions), and we can statically type the properties being passed to the static initialization functions.

### CODE REVIEW

File: client-js/app/scripts/app.ts [hunk 1]
```diff
@@ -141,20 +153,16 @@ window.addEventListener('WebComponentsReady', function() {
...
-  function loadDataset(path: Array<string>, dataset: String) {
+  function loadDataset(path: Array<string>, dataset: string) {
     console.log('loading dataset:', path, dataset);
     dv.dir(path).then((listing) => {
       app.route = 'dataset';
-      app.path = path;
-      app.dataset = dataset;
-      app.parentUrl = '/grapher/' + pathStr(path);
+      var parentUrl = '/grapher/' + pathStr(path);
+      setContent(pages.DatasetPage.init(path, dataset, parentUrl);
```
Comment (joshm-3b6b58) [inline client-js/app/scripts/app.ts]: Missing parenthesis

Comment (maffoo) [inline client-js/app/scripts/app.ts]: fixed.

**Comment by joshm-3b6b58** [comment]
The page renders differently than before, which is a bit concerning because I didn't see you changing any CSS:
This PR:
screenshot from 2015-09-22 15 22 43
Master:
screenshot from 2015-09-22 15 25 04

I'm fine with the aesthetic change if you meant to make it, but I'm concerned if there's something a wrong if the layout is changing unintentionally.

**Comment by maffoo** [comment]
The change was unintentional, but my ignoring style issues for now was intentional. The problem is that the paper-material elements were being styled by rules in app-theme.html and when I moved those elements into their own components to encapsulate the various pages, those rules are no longer getting applied, presumably because the paper-material element is now hidden inside the shadow DOM of the page element. I guess the autobinding template we were using before in index.html works differently than other custom elements and does not introduce a shadow DOM.

Polymer used to have a method to reach inside the shadow DOM and apply styles, but those mechanisms (the `::shadow` and `/deep/` css selectors) have been deprecated, and this area is currently in flux for web components generally (see [here](https://www.polymer-project.org/1.0/docs/devguide/styling.html#background) for some discussion).

**Comment by joshm-3b6b58** [comment]
I see. LGTM. 

I like this, the app is much more well behaved now.

**Pull request merged**